src/IdentityAuthClient.js

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import MatrixClientPeg from './MatrixClientPeg';
+
+export default class IdentityAuthClient {
+    constructor() {
+        this.accessToken = null;
+        this.authEnabled = true;
+    }
+
+    hasCredentials() {
+        return this.accessToken != null; // undef or null
+    }
+
+    // Returns a promise that resolves to the access_token string from the IS
+    async getAccessToken() {
+        if (!this.authEnabled) {
+            // The current IS doesn't support authentication
+            return null;
+        }
+
+        let token = this.accessToken;
+        if (!token) {
+            token = window.localStorage.getItem("mx_is_access_token");
+        }
+
+        if (!token) {
+            token = await this.registerForToken();
+            this.accessToken = token;
+            window.localStorage.setItem("mx_is_access_token", token);
+        }
+
+        try {
+            await this._checkToken(token);
+        } catch (e) {
+            // Retry in case token expired
+            token = await this.registerForToken();
+            this.accessToken = token;
+            window.localStorage.setItem("mx_is_access_token", token);
+        }
+
+        return token;
+    }
+
+    _checkToken(token) {
+        // TODO: Test current API token via /account endpoint
```

Comment:
Should this be resolved before merging?

Reply:
So far, the `/account` endpoint hasn't been implemented in Sydent. At the same time, the tokens don't seem to expire at the moment, but I suppose people could have bad tokens through some other means. I'll extend the comment for now. We'll also want to clear the token as part of future work where the IS can change during the session.